src/Runtime/Distributions/StringDistribution.cs

```diff
@@ -77,22 +77,21 @@ public static StringDistribution Char(DiscreteChar characterDist)
         /// <returns>The created distribution.</returns>
         public static StringDistribution CaseInvariant(string template)
         {
-            StringDistribution result = StringDistribution.Empty();
+            var result = StringAutomaton.Builder.Zero();
+            var last = result.Start;
             foreach (var ch in template)
             {
                 var upper = char.ToUpperInvariant(ch);
                 var lower = char.ToLowerInvariant(ch);
-                if (upper == lower)
-                {
-                    result.AppendInPlace(ch);
-                }
-                else
-                {
-                    result.AppendInPlace(DiscreteChar.OneOf(lower, upper));
-                }
+                var elem =
+                    upper == lower
+                        ? DiscreteChar.PointMass(lower)
+                        : DiscreteChar.OneOf(lower, upper);
+                last = last.AddTransition(elem, Weight.One);
```

Comment:
It seems clumsy to have to build up a StringDistribution by explicitly adding transitions to a StringAutomaton. Can we not have a StringDistribution builder which provides an Append etc?

Reply:
Actually, this function became 2 times faster compared to version before this PR. So working at the lower level may be more clumsy but is worth that. (And it showed up in profiling quite high).

Creating a special builder for string distributions is certainly possible. I'm not sure if it will make things more clear, but I will try.

Reply:
Someone creating string distributions should not have to know that they are implemented as automata. So the API for string distributions should reflect that. The time improvements are very welcome, but I think it is important not to degrade the API.

Reply:
Ok, I decided against building generic builder. And added new constructor into SequenceDistribution class: Consecutive(IEnumerable<TElementDistribution> sequence). It is way easier than adding some useful builder and satisfies all our use-cases so far. Slightly less efficient, but whatever.

Not sure if it is a good name for constructor.

Reply:
OK, I understand that the automaton builder is much more general than what we might want as a StringDistribution builder. The use case, especially with transducers, is that we build up pieces of the desired Distribution or Transducer (often using Append), and then combine the pieces in different ways (again, often using append). I guess your proposed constructor or (preferred) factory method will enable this. I don't like 'Consecutive'. Perhaps 'Concatenation'.